rust/src/data_structures.rs

```diff
@@ -323,10 +327,10 @@ pub enum TransportTuple {
 }
 
 impl TransportTuple {
-    /// Local IP address.
-    pub fn local_ip(&self) -> IpAddr {
-        let (Self::WithRemote { local_ip, .. } | Self::LocalOnly { local_ip, .. }) = self;
-        *local_ip
+    /// Local IP address or hostname.
+    pub fn local_address(&self) -> String {
```

Comment:
@nazar-pc should this return  `-> &String` instead (and hence replace `*local_address` with `local_address`? More fixed are needed no matter which way we go here, however I expect that returning a string reference is preferred over returning a copied one.

Reply:
I think `&str` makes most sense, for some use cases you don't need an owned value and will avoid allocation that way.